Mlem/API/Requests/CreateCommentLikeRequest.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  CreateCommentLikeRequest.swift
+//  Mlem
+//
+//  Created by Nicholas Lawson on 09/06/2023.
+//
+
+import Foundation
+
+struct CreateCommentLikeRequest: APIRequest {
+    
+    typealias Response = CreateCommentLikeResponse
+    
+    let endpoint: URL
+    let method: HTTPMethod
+    
+    struct Body: Encodable {
+        let auth: String
+        let comment_id: Int
+        let score: Int
+    }
+    
+    init(
+        account: SavedAccount,
+        commentId: Int,
+        score: ScoringOperation
+    ) throws {
+        do {
+            let data = try JSONEncoder().encode(
+                Body(
+                    auth: account.accessToken,
+                    comment_id: commentId,
+                    score: score.rawValue
+                )
+            )
+            self.endpoint = account.instanceLink
+                .appending(path: "comment")
+                .appending(path: "like")
```

Comment:
I'm also not sure if this is the most idiomatic way of declaring an API's path

Is there a way to define associated constants per structs? This way, the protocol just has to state there has to be a path string, which then gets added onto the instance via this, in an extension method or something, reducing more boilerplate

Reply:
we could adjust the protocol to require a path string, and then move the constructing into a default implementation - that should help, I'll take a look

Reply:
addressed by dbfe9740835ee1db247427226641e3617925b840